src/components/shared/changeBalanceForm/ChangeBalanceForm.tsx

```diff
@@ -0,0 +1,107 @@
+import { FC, useState } from 'react';
+
+import { Stack, Text, Group, Button, NumberInput } from '@mantine/core';
+import { useForm } from '@mantine/form';
+import { modals } from '@mantine/modals';
+import { IconCoins, IconArrowBackUp } from '@tabler/icons-react';
+import { useTranslation } from 'react-i18next';
+
+import { BalanceOperation } from 'constant';
+import { useAppDispatch, useAppSelector } from 'hooks';
+import { getNickname, setBalanceHistoryThunk, getCurrentBalance } from 'store';
+
+interface IProps {
+  userId: string;
+}
+
+export const ChangeBalanceForm: FC<IProps> = ({ userId }) => {
+  const dispatch = useAppDispatch();
+  const userNickname = useAppSelector(getNickname);
+  const currentBalance = useAppSelector(getCurrentBalance);
+
+  const { t } = useTranslation();
+
+  const [operationType, setOperationType] = useState<Nullable<BalanceOperation>>(null);
+
+  const form = useForm({
+    validateInputOnChange: ['amount'],
+    validate: {
+      amount: (value) => (value <= 0 ? t('errorAmount') : null),
+    },
+  });
+
+  const onSubmitHandler = (value: { amount: number }): void => {
+    const values = {
+      currentBalance,
+      amount: value.amount,
+      performerName: userNickname,
+      dateOperation: new Date(),
+      typeOperation: operationType,
+    };
+
+    const data = {
+      id: userId,
+      values,
+    };
+
+    if (userId && operationType) {
+      dispatch(setBalanceHistoryThunk(data)).then(() => {
+        setOperationType(null);
+        form.reset();
+        modals.closeAll();
+      });
+    }
+  };
+
+  return (
+    <form onSubmit={form.onSubmit((value) => onSubmitHandler(value))}>
```

Comment:
try to remove this - `(value) =>...(value))`
```suggestion
    <form onSubmit={form.onSubmit(onSubmitHandler)}>
```